### PR TITLE
Modified to check the expiration time of the allowlist when determining the vulnerability of an artifact

### LINKED ISSUE #18078: Ask about the expiration time of the CVE whitelists
**How can we help you?**

Hi,

I am inquiring about the expiration time of the CVE whitelists.
Harbor version is Version v2.5.3-797c3536.

I set the system wide CVE whitelists and expiration time to pull image with vulnerability. And I do not set per-project CVE whitelists and expiration time.
It seems that the whitelist is applied even after the expiration time has passed. Please check if this is the correct operation or a bug.

If this is the correct operation, shouldn't the whitelist be allowed after the expiration time?

Thanks.

### PR DESCRIPTION
**Pull request by hyeyoung-leee**
The allowlist is applied even after the expiration time has passed.

Signed-off-by: hyeyoung-leee hyeyoung.lee@nhn.com

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #18078 

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ x ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Review by zyyw — APPROVED** [review]
lgtm

File: src/controller/scan/base_controller.go [hunk 1]
```diff
@@ -815,18 +815,26 @@ func (bc *basicController) GetVulnerable(ctx context.Context, artifact *ar.Artif
...
-				vulnerable.CVEBypassed = append(vulnerable.CVEBypassed, v.ID)
+		if allowlistIsExpired {
+			for _, v := range vuls {
+				if severity == "" || v.Severity.Code() > severity.Code() {
+					severity = v.Severity
+				}
+			}
+		} else {
+			for _, v := range vuls {
+				if allowlist.Contains(v.ID) {
```
Comment (chlins) [inline src/controller/scan/base_controller.go:826]: If change this line to `if !allowlistIsExpired && allowlist.Contains(v.ID) {}`, then line 818-823 can be removed as logic duplicated with line 835-837?

Comment (zyyw) [reply src/controller/scan/base_controller.go:826]: @hyeyoung-leee could you please update the code to resolve this comment? thanks

Comment (hyeyoung-leee) [reply src/controller/scan/base_controller.go]: @chlins @zyyw 
Thanks for the comment.
I updated the code based on the comment.

**Review by chlins — APPROVED** [review]
lgtm

**Comment by OrlinVasilev** [comment]
@hyeyoung-leee Congratulations on your first merged PR on Harbor :) Welcome to the Club :)

**Pull request merged**